lang/mrustc/Portfile

```diff
@@ -59,46 +62,276 @@ compiler.blacklist *gcc*
+    depends_build-append port:python27
 }
 
+openssl.branch  1.0
+
+set legacy_features ""
+set legacy_cmake_opts ""
+
+# macOS 10.7 required to use MacPorts clang with emulated TLV
+if {${os.platform} eq "darwin" && ${os.major} < 11} {
+    # __cxa_thread_atexit is part of libcxx, link against it!
+    configure.cxx_stdlib \
+                libc++
+
+    patchfiles-append \
+                0009-macOS-10.6-emulated-TLS-before-10.7.patch \
+                0010-macOS-10.6-fix-build-issues-pre-Lion.patch \
+                0011-macOS-10.6-fix-pthread_cond_destroy.patch \
+                0012-macOS-10.6-support-legacy-crypto.patch
+
+    set legacy_features ",backtrace/std"
+
+    depends_build-append \
+                port:libtool
+
+    set legacy_cmake_opts " CMAKE_LIBTOOL=${prefix}/bin/libtool"
+
+}
+
+# #See: https://github.com/thepowersgang/mrustc/issues/225
+# use_parallel_build no
+
+# To make things easy keep all build envs global
+build.env       CC=${configure.cc} \
+                CXX=${configure.cxx} \
+                CFG_DEFAULT_LINKER=${configure.cc} \
+                "CXXFLAGS_EXTRA=\"${configure.cxxflags}\"" \
+                "LINKFLAGS_EXTRA=\"${configure.ldflags}\"" \
+                OPENSSL_DIR=[openssl::install_area] \
+                OPENSSL_INCLUDE_DIR=[openssl::include_dir] \
+                OPENSSL_LIB_DIR=[openssl::lib_dir] \
+                RUSTC_VERSION=${rust_version} \
+                MRUSTC_TARGET_VER=${rust_version_major} \
+                "MINICARGO_FLAGS=--features curl-sys/force-system-lib-on-osx${legacy_features}" \
+                LIBGIT2_SYS_USE_PKG_CONFIG=1 \
+                OUTDIR_SUF=-${rust_version} \
+                "LLVM_CMAKE_OPTS_EXTRA=PYTHON_EXECUTABLE=${pythonfullpath}${legacy_cmake_opts}"
+
+# Step 0: let start from building mrustc and its libds
+build.target    -f minicargo.mk bin/mrustc LIBS
+
 destroot {
-    xinstall -m 0755 -W ${worksrcpath} ./bin/${name} ${destroot}${prefix}/bin
-    xinstall -m 0755 -W ${worksrcpath} ./bin/minicargo ${destroot}${prefix}/bin
-    file copy ${worksrcpath}/output-${rust_version} ${destroot}${prefix}/lib/mrustc-${rust_version_major}
+    xinstall -m 0755 ${worksrcpath}/bin/mrustc ${destroot}${prefix}/bin
+
+    # install all .rlib and .rlib.hir to lib/
+    xinstall -d ${destroot}${prefix}/lib/mrustc-${rust_version_major}
+    foreach f [glob -tails -directory ${worksrcpath}/output-${rust_version} lib*.rlib lib*.rlib.hir lib*.rlib.c lib*.rlib.o] {
+        xinstall -m 0644 -W ${worksrcpath}/output-${rust_version} $f \
+            ${destroot}${prefix}/lib/mrustc-${rust_version_major}
+    }
 }
 
-# Run a Hello, world! test
-test.env            RUSTC_VERSION=${rust_version} MRUSTC_LIBDIR=output-${rust_version}
-test.run            yes
-test.target         test
+# Step 1: now I need minicargo
+subport minicargo {
+    description     Alternative implementation of Rust cargo
+    long_description {*}${description}
```

Comment:
Should mention in the long description that the command line interface is not compatible with regular cargo (unlike mrustc which mostly accepts the same flags as rustc).

Reply:
Seem that minicargo is only really used to bootstrap mrustc-cargo

Instead change the description to explain minicargo is simply used to bootstrap mrustc-cargo

Reply:
@evanmiller / @Gcenx nope, it is also used to build rustc. Let me rephase this a bit.